### PR TITLE
Improved Vagrant provider detection and support for parallels provider.

### PR DESCRIPTION
**Pull request by rickard-von-essen**
Improved the detection of Vagrant provider. The previous didn't
correctly detect which provider was used.

Added support for detecting the state of Parallels vagrant provider.

### CODE REVIEW

**Comment by rickard-von-essen** [comment]
Closes #53

File: fish_prompt.fish [hunk 1]
```diff
@@ -238,23 +238,32 @@ end
 
 function __bobthefish_prompt_vagrant -S -d 'Display Vagrant status'
   [ "$theme_display_vagrant" = 'yes' -a -f Vagrantfile ]; or return
-  if type -q VBoxManage
-    __bobthefish_prompt_vagrant_vbox
-  else if grep vmware_fusion Vagrantfile >/dev/null ^&1
-    __bobthefish_prompt_vagrant_vmware
+  set provider (__bobthefish_vagrant_provider)
```
Comment (bobthecow) [inline fish_prompt.fish]: Let's use `set -l` here.

File: fish_prompt.fish [hunk 2]
```diff
@@ -238,23 +238,32 @@ end
...
+    case 'vmware_fusion'
+      __bobthefish_prompt_vagrant_vmware
+    case 'parallels'
+      __bobthefish_prompt_vagrant_parallels
   end
 end
 
-function __bobthefish_vagrant_ids -S -d 'List Vagrant machine ids'
-  for file in .vagrant/machines/**/id
+function __bobthefish_vagrant_provider -S -d 'Get (first) Vagrant provider'
```
Comment (bobthecow) [inline fish_prompt.fish]: To be honest, I only use one Vagrant provider, so I've never tested how it works when you have multiple. Is it possible for there to be VMs from two or more Vagrant providers in the same directory? If so, is there any reason you'd care about all of them?

**Comment by rickard-von-essen** [comment]
I have improved this some more. I'll supply some more code later today.

**Comment by rickard-von-essen** [comment]
Now it should work for all mixed provider cases.

**Comment by bobthecow** [comment]
Looking good! It should be optimized a bit for performance, but I'll add that in a separate commit.

**Comment by bobthecow** [comment]
I optimized it a bit in 8898658 and was able to speed up the time spent on bobthefish's by 10-15x :)

Thanks again for implementing this!

**Pull request merged**